### PR TITLE
Add docs for the new Okta Expression Language used in OIE.

### PR DESCRIPTION
**Pull request by chenningzhang-okta**
## Description:
- **What's changed?** Add docs for the new Okta Expression Language used in OIE
- **Is this PR related to a Monolith release?** No. This PR is ready to be merged.

### Resolves:

* [OKTA-371557](https://oktainc.atlassian.net/browse/OKTA-371557)

@nghadge-okta @daytonwilliams-okta

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-oie/index.md [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+> **Note:** For the following expression examples, assume that *user* is a member of the following groups.
+
+| Group ID                 | Group Name               | Group Type            |
+| --------                 | -----------              | -----------           |
+| 00gak46y5hydV6NdM0g4     | Everyone                 | BUILT_IN              |
+| 00g1emaKYZTWRYYRRTSK     | West Coast Users         | OKTA_GROUP            |
+| 00garwpuyxHaWOkdV0g4     | West Coast Admins        | OKTA_GROUP            |
+| 00gjitX9HqABSoqTB0g3     | Engineering Users        | APP_GROUP             |
+
+Group functions take in a list of search criteria as input. Each search criteria is a key-value pair:<br>
```
Comment (daytonwilliams-okta) [inline packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-oie/index.md:144]: Again, why the `<br>`s?

File: packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-oie/index.md [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+Key: Specifies the matching attribute. Currently supported keys are `group.id`, `group.type`, and `group.profile.name`.<br>
+Value: Specifies a list matching values which can be exact values or a regex pattern (Only start with is supported)
```
Comment (daytonwilliams-okta) [inline packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-oie/index.md:146]: Nit:  Specifies a list `of` matching values ...

Nit ... a regex pattern (only supporting the [wildcard] wildcard to match `starts with`)

(Or something similar.  We should explicitly spell out exactly which wildcard(s) are supported, and how they're constrained, and the phrasing could use a tweak.)

File: packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-oie/index.md [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+* Function: `user.getLinkedObject($primaryName)`
+    * Parameter: (String primaryName)
+    * Return Type: User
+    * Example: `user.getLinkedObject("manager").lastName`
+    * Example Result: `Gates`
+
+### Time Functions
+
+> **Note:** For the following expression examples, assume that current date time is `2015-07-31T17:18:37.979Z`.
```
Comment (daytonwilliams-okta) [inline packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-oie/index.md:169]: Nit: assume `the` current date `and` time

File: packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-oie/index.md [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+| ----------------                                                                            | --------                                                    |
+| Refer to a `String` constant                                                                | `'Hello world'`                                             |
+| Refer to a `Integer` constant                                                               | `1234`                                                      |
+| Refer to a `Number` constant                                                                | `3.141`                                                     |
+| Refer to a `Boolean` constant                                                               | `true`                                                      |
+| Concatenate two strings                                                                     | `user.profile.firstName + user.profile.lastName`            |
+
+## Conditional Expressions
+
+<br>There are several rules for specifying the condition.
```
Comment (daytonwilliams-okta) [inline packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-oie/index.md:210]: Nit: The following rules apply to conditional expressions.

File: packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-oie/index.md [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+
+* Expressions must have valid syntax.
+* Expressions must evaluate to Boolean.
+* Expressions cannot contain an assignment operator, such as =.
+* User attributes used in expressions can contain only available User attributes.
```
Comment (daytonwilliams-okta) [inline packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-oie/index.md:215]: I'm sorry, I'm not following.  What does this mean?

I think it means something like `User attributes referenced in an expression must exist`?

File: packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-oie/index.md [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+
+| Expression                                                       | Output                                 |
+| -----------                                                      | -------                                |
+| `user.profile.country == "United States"`                        | True                                   |
+| `user.profile.intArray.contains(0)`                              | False                                  |
+| `user.profile.isContractor &#124;&#124; user.created.withinSeconds(0)`     | False                                  |
+
+You can use the ternary operator for performing IF...THEN...ELSE conditional logic inside the expression.
+
+
```
Comment (daytonwilliams-okta) [inline packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-oie/index.md:237]: Nit: extra blank line.

File: packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-oie/index.md [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+The format for conditional expressions is
```
Comment (daytonwilliams-okta) [inline packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-oie/index.md:238]: I think this should read `The format for a ternary contitional expression is`?

**Review by daytonwilliams-okta — COMMENTED** [review]
Very thorough, but I think there's a few places that could use some cleaning up.

**Review by nghadge-okta — COMMENTED** [review]
We looked at our original design doc to make sure everything is covered in this doc right?
Also for all these expressions have you tried running func test and make sure they are accurate?

File: packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-oie/index.md [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+|                                          |             | `user.profile.countryAlpha3.parseCountryCode()`                     | US (CountryCode object)   |
+|                                          |             | `'840'.parseCountryCode()`                                          | US (CountryCode object)   |
+| `$country_code_object.toAlpha2`          | String      | `'USA'.parseCountryCode().toAlpha2()`                               | "US"                      |
+| `$country_code_object.toAlpha3`          | String      | `'840'.parseCountryCode().toAlpha3()`                               | "USA"                     |
+| `$country_code_object.toNumeric`         | String      | `'United States'.parseCountryCode().toNumeric()`                    | "840"                     |
+| `$country_code_object.toName`            | String      | `'US'.parseCountryCode().toName()`                                  | "United States"           |
+
+**Note:**  All these functions can be called on ISO 3166-1 2-character country codes (Alpha 2), 3-character country codes (Alpha 3), numeric country codes, and country code Strings.
```
[the remark this answers is not in the document]
Comment (chenningzhang-okta) [reply packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-oie/index.md]: Rephrased this note. PTAL.

**Comment by jakubvul-okta** [comment]
@chenningzhang-okta please fill out the PR. Specifically which Monolith release this is related to. If you don't know, "TBD" is also acceptable.

File: packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-identity-engine/index.md [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+|                                          |             | `'840'.parseCountryCode()`                                          | US (CountryCode object)   |
+| `$country_code_object.toAlpha2`          | String      | `'USA'.parseCountryCode().toAlpha2()`                               | "US"                      |
+| `$country_code_object.toAlpha3`          | String      | `'840'.parseCountryCode().toAlpha3()`                               | "USA"                     |
+| `$country_code_object.toNumeric`         | String      | `'United States'.parseCountryCode().toNumeric()`                    | "840"                     |
+| `$country_code_object.toName`            | String      | `'US'.parseCountryCode().toName()`                                  | "United States"           |
+
+**Note:**  The `parseCountryCode` function can be called on the String representations of ISO 3166-1 2-character country codes (Alpha 2), 3-character country codes (Alpha 3), numeric country codes, and names of country.
+The other four functions can be called on country code object, and return the output in the format specified by the function names.
```
Comment (daytonwilliams-okta) [inline packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-identity-engine/index.md:134]: Nit: on `a` country code object.

File: packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-identity-engine/index.md [hunk 10]
```diff
@@ -0,0 +1,252 @@
...
+|                                          |             | `user.profile.countryAlpha3.parseCountryCode()`                     | US (CountryCode object)   |
+|                                          |             | `'840'.parseCountryCode()`                                          | US (CountryCode object)   |
+| `$country_code_object.toAlpha2`          | String      | `'USA'.parseCountryCode().toAlpha2()`                               | "US"                      |
+| `$country_code_object.toAlpha3`          | String      | `'840'.parseCountryCode().toAlpha3()`                               | "USA"                     |
+| `$country_code_object.toNumeric`         | String      | `'United States'.parseCountryCode().toNumeric()`                    | "840"                     |
+| `$country_code_object.toName`            | String      | `'US'.parseCountryCode().toName()`                                  | "United States"           |
+
+**Note:**  The `parseCountryCode` function can be called on the String representations of ISO 3166-1 2-character country codes (Alpha 2), 3-character country codes (Alpha 3), numeric country codes, and names of country.
```
Comment (daytonwilliams-okta) [inline packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-identity-engine/index.md:133]: Nit: country names.

File: packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-oie/index.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (daytonwilliams-okta) [reply packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-oie/index.md]: A few new nits.

**Review by daytonwilliams-okta — APPROVED** [review]
Looks good now, except for the two minor grammar tweaks introduced in the reword in the update.  I'll go ahead and approve it now :rocket: :sheep: :it:

**Review by nghadge-okta — APPROVED** [review]

File: packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-oie/index.md [hunk 11]
```diff
@@ -0,0 +1,247 @@
...
+meta:
+  - name: description
+    content: Learn more about the features and syntax of the Okta Expression Language in Okta Identity Engine.
+---
+
+# Overview
+
+Expressions allow you to reference, transform, and combine attributes before you store them on a user profile or before passing them to an application for authentication or provisioning. For example, you might use a custom expression to create a username by stripping @company.com from an email address. Or, you might combine firstName and lastName attributes into a single displayName attribute.
+
+This document details the features and syntax of the Okta Expression Language used in Okta Identity Engine. Expressions being used outside of Okta Identity Engine should continue using the features and syntax of [the legacy Okta Expression Language](/docs/reference/okta-expression-language/). This document will be updated over time as new capabilities
```
[the remark this answers is not in the document]
Comment (jakubvul-okta) [reply packages/@okta/vuepress-site/docs/reference/okta-expression-language-in-oie/index.md]: @chenningzhang-okta "Identity engine", capital "I".

**Comment by chenningzhang-okta** [comment]
Per PM request, this page is now hidden from nav bar and Google search.

Action items need to be done before this PR can be merged:
- https://oktainc.atlassian.net/browse/BUSAPPS-47421
- Add the correct `<ApiLifecycle>` annotation. Per @unniravindranathan-okta it should be `beta`, but @jakubvul-okta mentioned it should be `LGA` instead.

**Comment by chenningzhang-okta** [comment]
@susanharper-okta this PR is ready to be merged.

**Pull request merged**